File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/nu/newbabypicc/d/NewbabypiccAggDO.java
package iih.ci.mr.nu.newbabypicc.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.mr.nu.newbabypicc.d.desc.NewbabypiccAggDODesc;



/**
 * 新生儿科picc护理信息
 */
public class NewbabypiccAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public NewBabyPiccDO getParentDO() {
		return ((NewBabyPiccDO) super.getParentDO());
	}

	public void setParentDO(NewBabyPiccDO headDO) {
		setParent(headDO);
	}

	public NewBabyPiccRecDO[] getNewBabyPiccRecDO() {
		IBaseDO[] dos = getChildren(NewBabyPiccRecDO.class);
		if(dos==null) return null;
		NewBabyPiccRecDO[] result=new NewBabyPiccRecDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(NewBabyPiccRecDO)dos[i];
		}
		return result;
	}
	
	public void setNewBabyPiccRecDO(NewBabyPiccRecDO[] dos) {
		setChildren(NewBabyPiccRecDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		NewbabypiccAggDODesc desc = new NewbabypiccAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new NewBabyPiccDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.mr.nu.newbabypicc.d.NewBabyPiccRecDO")) {
             return new NewBabyPiccRecDO();
         }
         return null;
     }
}

File: client/iih_stub/iih.ci_stub/iih.ci.mr/nu/newbabypicc/d/NewbabypiccAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.mr.nu.newbabypicc.d
{
	/// <summary>
	/// 新生儿科picc护理信息DO
	/// </summary>
	public class NewbabypiccAggDO : BaseAggDO
	{	
	    private static string NEWBABYPICCDO_NEWBABYPICCRECDO="iih.ci.mr.nu.newbabypicc.d.NewBabyPiccRecDO";

        public NewbabypiccAggDO() {
            this.setParentDO(new NewBabyPiccDO());
        }

  	    public NewBabyPiccDO getParentDO() {
		    return ((NewBabyPiccDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(NewBabyPiccDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public NewBabyPiccRecDO[] getNewBabyPiccRecDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(NEWBABYPICCDO_NEWBABYPICCRECDO));
		    if (dos == null || dos.Length==0){return null;}
            NewBabyPiccRecDO[] r = new NewBabyPiccRecDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as NewBabyPiccRecDO;
            }
	        return r;
		    
	    }
	    
	    public void setNewBabyPiccRecDO(NewBabyPiccRecDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new NewBabyPiccDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (NEWBABYPICCDO_NEWBABYPICCRECDO == clzName)
	        {
                return new NewBabyPiccRecDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.mr.nu.newbabypicc.d.NewBabyPiccRecDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}
